src/components/swap/SwapBuyFiatButton.tsx

```diff
@@ -0,0 +1,144 @@
+import { MouseoverTooltipContent } from 'components/Tooltip'
+import { useWalletDrawer } from 'components/WalletDropdown'
+import { useCallback, useEffect, useState } from 'react'
+import { useBuyFiatFlowCompleted } from 'state/user/hooks'
+import styled from 'styled-components/macro'
+import { ExternalLink } from 'theme'
+
+import { useFiatOnrampAvailability, useOpenModal } from '../../state/application/hooks'
+import { ApplicationModal } from '../../state/application/reducer'
+
+const Dot = styled.div`
+  height: 8px;
+  width: 8px;
+  background-color: ${({ theme }) => theme.accentActive};
+  border-radius: 50%;
+`
+
+export const MOONPAY_REGION_AVAILABILITY_ARTICLE =
+  'https://support.uniswap.org/hc/en-us/articles/11306664890381-Why-isn-t-MoonPay-available-in-my-region-'
+
+enum BuyFiatFlowState {
+  // Default initial state. User is not actively trying to buy fiat.
+  INACTIVE,
+  // Buy fiat flow is active and region availability has been checked.
+  ACTIVE_POST_REGION_CHECK,
+  // Buy fiat flow is active, feature is available in user's region & needs wallet connection.
+  ACTIVE_NEEDS_WALLET_CONNECTION,
+}
+
+const StyledTextButton = styled(ButtonText)`
+  color: ${({ theme }) => theme.textSecondary};
+  gap: 4px;
+  &:focus {
+    text-decoration: none;
+  }
+  &:active {
+    text-decoration: none;
+  }
+`
+
+const POPOVER_DELAY_BEFORE_SHOW_MS = 500
+
+export default function SwapBuyFiatButton() {
+  // Importing useWeb3React here is necessary for mocking different behavior in
+  // different tests in SwapBuyFiatButton.test.tsx
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const { useWeb3React } = require('@web3-react/core')
+  const { account } = useWeb3React()
+  const openFiatOnRampModal = useOpenModal(ApplicationModal.FIAT_ONRAMP)
+  const [buyFiatFlowCompleted, setBuyFiatFlowCompleted] = useBuyFiatFlowCompleted()
+  const [checkFiatRegionAvailability, setCheckFiatRegionAvailability] = useState(false)
+  const {
+    available: fiatOnrampAvailable,
+    availabilityChecked: fiatOnrampAvailabilityChecked,
+    loading: fiatOnrampAvailabilityLoading,
+  } = useFiatOnrampAvailability(checkFiatRegionAvailability)
+  const [buyFiatFlowState, setBuyFiatFlowState] = useState(BuyFiatFlowState.INACTIVE)
+  const [walletDrawerOpen, toggleWalletDrawer] = useWalletDrawer()
+
+  /* 
+  Once the buy fiat flow has been completed, reset BuyFiatFlowState and also save to 
+  local storage that user has completed this flow already.
+  */
+  const terminateBuyFiatFlow = useCallback(() => {
+    setBuyFiatFlowCompleted(true)
+    setBuyFiatFlowState(BuyFiatFlowState.INACTIVE)
+  }, [setBuyFiatFlowCompleted, setBuyFiatFlowState])
+
+  /*
+  Depending on the current state of the buy fiat flow the user is in (buyFiatFlowState),
+  the desired behavior of clicking the 'Buy' button is different:
+  1) Initially upon first click, need to check the availability of the feature in the user's 
+  region, and continue the flow.
```

Comment:
nit: would it be possible to break this comment down, so that each point is placed right above `if` condition, so it's easier to scan?

Reply:
I tried breaking it into the separate lines and I feel like it looks messier? :S its in order that matches the is / else if statements anyway so I woud prefer keeping it like this.